
loading train data:    : 0it [00:00, ?it/s]

loading train data:    : 1839it [00:02, 919.12it/s]
loading train data:    : 3361it [00:08, 401.18it/s]
loading test data:    : 856it [00:00, 1948.80it/s]
Training:    :   0%|          | 0/160 [00:00<?, ?it/s]..\evaluation\eval.py:31: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  a = (((batch[student][:, 0:C.NUM_OF_QUESTIONS] - batch[student][:, C.NUM_OF_QUESTIONS:]).sum(1) + 1)//2)[1:]
> d:\ethz\internship\adaptive-e-learning-for-educational-recommendation-system\deepknowledgetracing-dkt-pytorch\dkt\knowledgetracing\evaluation\eval.py(48)train_epoch()
-> optimizer.zero_grad()
(Pdb) Adagrad (
Parameter Group 0
    eps: 1e-10
    initial_accumulator_value: 0
    lr: 0.002
    lr_decay: 0
    weight_decay: 0
)
(Pdb) DKT(
  (rnn): RNN(248, 200, batch_first=True)
  (fc): Linear(in_features=200, out_features=124, bias=True)
  (sig): Sigmoid()
)
(Pdb) *** TypeError: type() missing 1 required positional argument: 'dst_type'
(Pdb) <bound method Module.type of DKT(
  (rnn): RNN(248, 200, batch_first=True)
  (fc): Linear(in_features=200, out_features=124, bias=True)
  (sig): Sigmoid()
)>
(Pdb) *** AttributeError: 'DKT' object has no attribute 'device'
(Pdb) tensor([1789.6183], device='cuda:0', grad_fn=<SubBackward0>)
(Pdb)
Training:    :   0%|          | 0/160 [1:45:53<?, ?it/s]
Traceback (most recent call last):
  File "D:\ETHz\Internship\adaptive-e-learning-for-educational-recommendation-system\DeepKnowledgeTracing-DKT-Pytorch\DKT\KnowledgeTracing\evaluation\run.py", line 41, in <module>
    model, optimizer, train_loss = eval.train(trainLoaders, model, optimizer_adgd, loss_func,device)
  File "..\evaluation\eval.py", line 90, in train
    model, optimizer, loss = train_epoch(model, trainLoaders[i], optimizer, lossFunc,device)
  File "..\evaluation\eval.py", line 48, in train_epoch
    optimizer.zero_grad()
  File "..\evaluation\eval.py", line 48, in train_epoch
    optimizer.zero_grad()
  File "C:\Users\tyu06\anaconda3\lib\bdb.py", line 88, in trace_dispatch
    return self.dispatch_line(frame)
  File "C:\Users\tyu06\anaconda3\lib\bdb.py", line 112, in dispatch_line
    self.user_line(frame)
  File "C:\Users\tyu06\anaconda3\lib\pdb.py", line 262, in user_line
    self.interaction(frame, None)
  File "C:\Users\tyu06\anaconda3\lib\pdb.py", line 357, in interaction
    self._cmdloop()
  File "C:\Users\tyu06\anaconda3\lib\pdb.py", line 322, in _cmdloop
    self.cmdloop()
  File "C:\Users\tyu06\anaconda3\lib\cmd.py", line 126, in cmdloop
    line = input(self.prompt)
  File "C:\Users\tyu06\anaconda3\lib\codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte